.gitignore

```diff
@@ -3,6 +3,9 @@ node_modules
 /build
 /coverage
 packages/*/dist
+packages/**/*.js
```

Comment:
Do we really need this line? Won't it be covered by the `*/dist` rule above?

Reply:
the build:esmodules task actually builds `.js` files in the root of each package/component.

Reply:
Ahh ok. Would it be possible to generate the `.js` files in `dist/` or some other subdirectory so that we can more easily isolate them?

Reply:
well there are 2 versions of the .d.ts and .js files. There is the UMD and the ES6 versions. All the ES6 versioned code I've placed in the root, so when users want to import it will look like:

```
import Foo from @material/foo/index
```

But maybe we can put it in a `/esm` directory? So the entire package will look like:
```
/dist
/esm
package.json
...
```

WDYT?

Reply:
The idea was to keep the ESM JS files where they currently are, so that we don't break literally everybody currently using them. That's why the JS and corresponding d.ts files are directly under each package folder.

I knew it'd be a little rough to set up, but it looks like at the very least the gitignore and clean script are handling it here.

Reply:
I guess to follow that up: I'm assuming that d.ts should generally appear where the js files appear... if it's permissible to do otherwise, we could put the d.ts files for individual ESM files under dist or something (given that it seems like typescript just looks for .d.ts anywhere under the project), but that might seem confusing?

Reply:
I think it is convention that the `.d.ts` files appear sibling to the `.js` files, so I think it'll be confusing for people. I think we should stick to convention.

So it sounds like we need to keep the gitignore like this to keep backwards compatibility.